components/nexusoperations/workflow/commands.go

```diff
@@ -216,27 +216,22 @@ func (ch *commandHandler) HandleCancelCommand(
 		if errors.Is(err, hsm.ErrStateMachineNotFound) {
 			return workflow.FailWorkflowTaskError{
 				Cause:   enumspb.WORKFLOW_TASK_FAILED_CAUSE_BAD_REQUEST_CANCEL_NEXUS_OPERATION_ATTRIBUTES,
-				Message: fmt.Sprintf("requested cancelation for a non-existing operation with scheduled event ID of %d", attrs.ScheduledEventId),
-				// TODO(bergundy): Message: fmt.Sprintf("requested cancelation for a non-existing or already completed operation with scheduled event ID of %d", attrs.ScheduledEventId),
+				Message: fmt.Sprintf("requested cancelation for a nonexistent or terminated operation with scheduled event ID %d", attrs.ScheduledEventId),
 			}
 		}
 		return err
 	}
-	// TODO(bergundy): Remove this when operation auto-deletes itself on terminal state.
-	// Operation may already be in a terminal state because it doesn't yet delete itself. We don't want to accept
-	// cancelation in this case.
+
 	op, err := hsm.MachineData[nexusoperations.Operation](node)
 	if err != nil {
 		return err
 	}
-	// The operation is already in a terminal state and the terminal NexusOperation event has not just been buffered.
-	// We allow the workflow to request canceling an operation that has just completed while a workflow task is in
-	// flight since it cannot know about the state of the operation.
-	// TODO(bergundy): When we support state machine deletion, this condition will have to change.
-	if !nexusoperations.TransitionCanceled.Possible(op) && !ms.HasAnyBufferedEvent(makeNexusOperationTerminalEventFilter(attrs.ScheduledEventId)) {
```

Comment:
You will need allow cancel while a terminal event is buffered (e.g. the SDK isn't aware yet that the operation has completed).

Reply:
The check for buffered terminal event should be moved to where we error out when the node is not found.

Reply:
I think this is what you have in mind. Please take a look